pkg/sql/storageparam/tablestorageparam/table_storage_param.go

```diff
@@ -227,13 +237,41 @@ var tableParams = map[string]tableParam{
 			return nil
 		},
 		onReset: func(po *Setter, evalCtx *eval.Context, key string) error {
-			return errors.WithHintf(
-				pgerror.Newf(
+			if po.tableDesc.RowLevelTTL != nil {
+				po.tableDesc.RowLevelTTL.DurationExpr = ""
+			}
+			return nil
+		},
+	},
+	`ttl_expiration_expression`: {
+		onSet: func(ctx context.Context, po *Setter, semaCtx *tree.SemaContext, evalCtx *eval.Context, key string, datum tree.Datum) error {
+			stringVal, err := paramparse.DatumAsString(evalCtx, key, datum)
+			if err != nil {
+				return err
+			}
+			stringVal = strings.TrimSpace(stringVal)
+			if !strings.HasPrefix(stringVal, "(") && !strings.HasSuffix(stringVal, ")") {
+				stringVal = "(" + stringVal + ")"
+			}
+			// todo(wall): add type checking
+			_, err = parser.ParseExpr(stringVal)
+			if err != nil {
+				return pgerror.Wrapf(
+					err,
 					pgcode.InvalidParameterValue,
-					`resetting "ttl_expire_after" is not permitted`,
-				),
-				"use `RESET (ttl)` to remove TTL from the table",
-			)
+					`value of %q must be a valid expression`,
```

Comment:
can you explicitly mention here this si `ttl_expression`

Reply:
the %q changes here fine, but %q is not always what you want as it escapes `"` (which may not be what you want)
ah this does reference the key, nvm